convert2rhel/actions/__init__.py

```diff
@@ -269,14 +279,49 @@ class ActionMessageBase:
     :type id: str
     :keyword level: The level to be set, defaulting to SUCCESS.
     :type: level: str
-    :keyword message: The message to be set.
-    :type message: str | None
+    :keyword title: The title to be set.
+    :type title: str
+    :keyword description: The description to be set.
+    :type description: str
+    :keyword diagnosis: The diagnosis to be set.
+    :type diagnosis: str | None
+    :keyword remediation: The remediation to be set.
+    :type remediation: str | None
     """
 
-    def __init__(self, level="SUCCESS", id=None, message=""):
+    def __init__(self, level="SUCCESS", id="SUCCESS", title="", description="", diagnosis="", remediation=""):
         self.id = id
         self.level = STATUS_CODE[level]
-        self.message = message
+        self.title = title
+        self.description = description
+        self.diagnosis = diagnosis
+        self.remediation = remediation
+
+    def __eq__(self, other):
+        if (
+            self.level == other.level  # pylint: disable=too-many-boolean-expressions
+            and self.id == other.id
+            and self.title == other.title
+            or self.description == other.description
+            or self.diagnosis == other.diagnosis
+            or self.remediation == other.remediation
+        ):
```

Comment:
Can we in this instance have something with a different level, id and title but same description and it would be the same?

Reply:
I think, as I think Eric is getting at too, that we want all of the fields in `self` to be equal to all of the equivalent fields in `other` for this to evaluate as equal.

Reply:
Oh!  Also, I think we implemented `__eq__()`  before we implemented `__hash__()`.  Since we do have `__hash__()` now, it would be more maintainable to just use `hash()` here:
```suggestion
        if hash(self) == hash(other):
```

Reply:
Yes `__eq__` did come before `__hash__`, I agree with this change